src/display.c

```diff
@@ -130,11 +157,53 @@ int display_parse_mode(const char *config, dcp_timing_mode_t *mode)
     return mode->valid;
 }
 
+int display_wait_connected(dcp_iboot_if_t *iboot, int *timing_cnt, int *color_cnt)
+{
+    int hpd;
+
+    for (int retries = 0; retries < DISPLAY_STATUS_RETRIES; retries += 1) {
+        hpd = dcp_ib_get_hpd(iboot, timing_cnt, color_cnt);
+        if ((hpd > 0) && *timing_cnt && *color_cnt) {
+            printf("display: waited %d ms for display connected\n", retries * DISPLAY_STATUS_DELAY);
+            return 1;
+        }
+
+        mdelay(DISPLAY_STATUS_DELAY);
+    }
+
+    // hpd is 0 if no display, negative if an error occurred
+    return hpd;
+}
+
+int display_wait_disconnected(dcp_iboot_if_t *iboot, int wait_delay)
+{
+    int hpd, timing_cnt, color_cnt;
+    int max_retries = wait_delay * 1000 / DISPLAY_STATUS_DELAY;
+
+    for (int retries = 0; retries < max_retries; retries += 1) {
+        hpd = dcp_ib_get_hpd(iboot, &timing_cnt, &color_cnt);
```

Comment:
`dcp_ib_get_hpd` can be called with `NULL` for timing_cnt and color_cnt, would be simpler here since we are just waiting for `hpd == 0`